cli/src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala

```diff
@@ -86,13 +86,23 @@ class AnalyseExecutor(toolInput: Option[String],
         }
     }
 
-    formatter.end()
-
-    analysisResult.map { result =>
+    val analysisResultsWithMissingMetrics = analysisResult.map { result =>
       val (metricsResults, issuesResults, duplicationResults) =
         result.partitionSubtypes[MetricsToolExecutorResult, IssuesToolExecutorResult, DuplicationToolExecutorResult]
-      MetricsToolExecutor.reduceMetricsToolResultsByFile(metricsResults) ++ issuesResults ++ duplicationResults
+      reduceAndCalculateMissingFileMetrics(metricsResults) ++ issuesResults ++ duplicationResults
     }
+
+    formatter.end()
+    analysisResultsWithMissingMetrics
+  }
+
+  private def reduceAndCalculateMissingFileMetrics(
```

Comment:
Try to put this inside some post processing logic and then perform the reduce, missing and format inside.